packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -72,9 +75,129 @@ export class StateManager {
       ...options,
     };
 
-    return this.getData<$TSTeamProviderInfo>(filePath, mergedOptions);
+    return this.getData<CustomIAMPolicies>(filePath, mergedOptions);
   };
 
+  getCustomPolicies = (categoryName: string, resourceName: string): any => {
+    const filePath = pathManager.getCustomPoliciesPath(categoryName, resourceName);
+    let customPolicies: CustomIAMPolicies = {
+      policies :  []
+    }
+    const data = JSONUtilities.readJson<CustomIAMPolicies>(filePath, {throwIfNotExist : false});
+    const ajv = new Ajv();
+    const { envName } = this.getLocalEnvInfo()
+
+    //validate if the policies match the custom IAM policies schema, if not, then not write into the CFN template
+    const validatePolicies = ajv.compile(CustomIAMPoliciesSchema);
+    if (!data || !validatePolicies(data)) {
+      printer.warn(`The format of custom IAM policies in the ${categoryName} ${resourceName} is not valid`);
+      return undefined;
+    }
+    const validatePolicy = ajv.compile(CustomIAMPolicySchema);
```

Comment:
This is good but why are you validating again?

Reply:
They are different, the other one is to see if policies exist, this one is to validate the custom IAM policy inside the policies.

Reply:
Doesn't the schema validate the full structure?

Reply:
I separate them into two schema for I have type CustomIAMPolicies and CustomIAMPolicy

Reply:
What's the difference between the two?

Reply:
CustomIAMPolicy is the array contains CustomIAMPolicies. I can just keep CustomIAMPolicies if this is confusing

Reply:
Can you see if it works in a single schema?

Reply:
Yes, it works